docs/element-definitions.md

```diff
@@ -234,6 +239,49 @@ Note: At the moment only the HTTP protocol is supported.
 }
 ```
 
+### Affordance (Base API Element)
+
+An affordance element provides a way to provide affordance definitions for the
+API. These definitions can describe link relations used in a Transition Element,
+along with defining unsafe and idempotent transitions.
+
+While data structures are used for defining semantics around data models,
+affordances are used for defining semantics around hypermedia state transitions.
+
+The control types used here are based on the types used in [ALPS][]. The meta
+property `id` SHOULD be used to define the machine-usable name of the
+affordance.
+
+#### Properties
+
+- `element`: affordance
+- `attributes`
+    - `controlType` (enum)
+        - safe - A hypermedia control that triggers a safe, idempotent state
+          transition (e.g. HTTP.GET or HTTP.HEAD).
+        - unsafe - A hypermedia control that triggers an unsafe, non-idempotent
+          state transition (e.g. HTTP.POST).
+        - idempotent - A hypermedia control that triggers an unsafe, idempotent
```

Comment:
I'm actually curious if there is something we could do differently with this to make it better. I think I agree with your concern here and think we should move away from it. I have a few thoughts on that along with yours, and I think it would be better to discuss specifically rather than just here in comments.

Thanks for bringing this up. Good catch on this.